### PR TITLE
README: update client examples

### LINKED ISSUE #545: Running Cockroach with Docker command in README doesn't work on Mac
I get this:

```
docker run -p 8080:8080 "cockroachdb/cockroach" init -rpc="localhost:0" -stores="ssd=$(mktemp -d)"
usage: mktemp [-d] [-q] [-t prefix] [-u] template ...
       mktemp [-d] [-q] [-u] -t prefix
E0402 04:54:15.937405       8 start.go:91] Failed to initialize context: context.go:131: invalid or empty engines specification "ssd="
```

### PR DESCRIPTION
**Pull request by tbg**
closes #545

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,29 +34,81 @@ See [TODO.md](https://github.com/cockroachdb/cockroach/blob/master/TODO.md)
...
+
+```
+# Put the values a->1, b->2, c->3, d->4.
+$ ./cockroach put a 1 b 2 c 3 d 4
+$ ./cockroach scan
+"a"     1
+"b"     2
+"c"     3
+"d"     4
+# Scans do not include the right endpoint.
```
Comment (petermattis) [inline README.md]: Maybe rephrase as "Scans do not include the end key".

File: README.md [hunk 2]
```diff
@@ -34,29 +34,81 @@ See [TODO.md](https://github.com/cockroachdb/cockroach/blob/master/TODO.md)
...
+root@82cb657cdc42:/cockroach#
 ```
-This bootstraps and starts a single node with one temporary RocksDB instance in the background (remove the `-d` flag if you want to see stdout).
-Now let's talk to this node. You can use the [REST Explorer at
-localhost:8080](http://localhost:8080) or talk directly to the API:
+
+Now we're in an environment that has everything set up, and we start by firing up a node:
+
+```bash
+$ ./cockroach init -http 0.0.0.0:8080 -stores ssd="$(mktemp -d /tmp/dbXXX)" &> node.log &
```
Comment (petermattis) [inline README.md]: Heh, wasn't aware of the `&>` redirection. Thought I knew bash.

Comment (petermattis) [inline README.md]: Do you need to specify the `-http` flag? Isn't the default value (`:8080`) sufficient?

**Comment by petermattis** [comment]
LGTM. We need to spend more time on documentation.

File: README.md (hunk 2 above)
Comment (tschottdorf) [inline README.md]: In the Docker container, the default will bind to one of the interfaces (172.*) only, not to localhost. Maybe we want to change the default?

Comment (petermattis) [inline README.md]: Huh, I'm surprised that the default will only bind to one of the interfaces. Yeah, I think changing the default makes sense.

Comment (tschottdorf) [inline README.md]: It's because of this:

```
	if strings.HasPrefix(httpAddr, ":") {
		httpAddr = s.host + httpAddr
	}
```

Feels like I should be able to just take this out without losing anything. What do you think @petermattis?

**Comment by tschottdorf** [comment]
HTTP server now listens on all interfaces with the default ":8080". README updated accordingly.
PTAL @petermattis - I wonder why the hostname was put there in the first place.

**Comment by petermattis** [comment]
I'm not sure what the hostname stuff was about. Perhaps @bdarnell or @spencerkimball does.

File: README.md [hunk 3]
```diff
@@ -34,29 +34,81 @@ See [TODO.md](https://github.com/cockroachdb/cockroach/blob/master/TODO.md)
...
 * `make build`
-* replace `docker [...] [run|build] [...]` by `./cockroach` below.
+* ignore the initial calls to `docker` below.
 
 #### Bootstrap and talk to a single node
 
+Getting a Cockroach node up and running is easy. If you have the `cockroach` binary, skip over the next shell session. Most users however will want to run the following:
+
 ```bash
-$ docker run -d -p 8080:8080 "cockroachdb/cockroach" \
```
Comment (treeder) [inline README.md]: Why get rid of this single liner?  This is a really nice way to get started.

Comment (treeder) [inline README.md]: Btw, if this is to fix #545, I just started it with the following and everything was fine:

```sh
docker run -p 8080:8080 "cockroachdb/cockroach" init -rpc="localhost:0" -stores="hdd=/tmp/coackroachdata"
```

Comment (tschottdorf) [inline README.md]: The fix to #545 is in line 61. You need to pass a template to `mktemp -d` on OSX.

Glad you liked the single liner - but if you're starting a node, and you want to use the cockroach-cli, then you better be in a shell on that node or you're going to have to pass the explicit location of the api endpoint around.

I'll see if there's another good place to put that line.

**Comment by bdarnell** [comment]
I suspect the hostname thing was done to avoid firewall prompts on the mac, but I'm not sure in this case. Regardless, I agree with removing it. If there are lingering firewall issues we should add a hostname where we need it instead of a global hack like this.

File: README.md [hunk 4]
```diff
@@ -91,6 +139,15 @@ Once you've built your image, you may want to run the tests:
 * `docker run "cockroachdb/cockroach-dev" test`
 * `make acceptance`
 
+Assuming you've built `cockroachdb/cockroach`, let's run a simple Cockroach node in the background:
```
Comment (tschottdorf) [inline README.md]: @treeder

Comment (treeder) [inline README.md]: Cool, thanks! :)
